lib/audited/auditor.rb

```diff
@@ -219,11 +219,28 @@ def audit_destroy
       end
 
       def write_audit(attrs)
+        return unless should_audit?(attrs)
         attrs[:associated] = send(audit_associated_with) unless audit_associated_with.nil?
         self.audit_comment = nil
         run_callbacks(:audit)  { audits.create(attrs) } if auditing_enabled
       end
 
+      def should_audit?(attrs)
+        if audited_options[:if]
+          case audited_options[:if]
+          when Proc
+            audited_options[:if].call(self, attrs)
+          when Symbol
+            send(audited_options[:if], attrs)
+          else
+            errors.add(:audit, "Unsupported conditional argument of type #{audited_options[:if].class}")
```

Comment:
Seems like this should raise an exception instead of adding an error. Errors are user-facing and a user can't do anything about it. Exceptions are programmer-facing.